code/modules/mob/dead/new_player/poll.dm

```diff
@@ -513,14 +518,21 @@
 	if(!replytext)
 		to_chat(usr, "The text you entered was blank. Please correct the text and submit again.")
 		return
-	var/adminrank = sanitizeSQL(poll_check_voted(pollid, TRUE))
+	var/voted = poll_check_voted(pollid, text = TRUE, silent = TRUE)
+	if(isnull(voted))
+		return
+	var/adminrank = sanitizeSQL(poll_rank())
 	if(!adminrank)
 		return
 	replytext = sanitizeSQL(replytext)
 	if(!(length(replytext) > 0) || !(length(replytext) <= 8000))
 		to_chat(usr, "The text you entered was invalid or too long. Please correct the text and submit again.")
 		return
-	var/datum/DBQuery/query_text_vote = SSdbcore.NewQuery("INSERT INTO [format_table_name("poll_textreply")] (datetime ,pollid ,ckey ,ip ,replytext ,adminrank) VALUES (Now(), [pollid], '[ckey]', INET_ATON('[client.address]'), '[replytext]', '[adminrank]')")
+	var/datum/DBQuery/query_text_vote
+	if(!voted)
+		query_text_vote  = SSdbcore.NewQuery("INSERT INTO [format_table_name("poll_textreply")] (datetime ,pollid ,ckey ,ip ,replytext ,adminrank) VALUES (Now(), [pollid], '[ckey]', INET_ATON('[client.address]'), '[replytext]', '[adminrank]')")
+	else
+		query_text_vote  = SSdbcore.NewQuery("UPDATE [format_table_name("poll_textreply")] SET datetime = Now(), ip = INET_ATON('[client.address]'), replytext = '[replytext]' WHERE pollid = '[pollid]' AND ckey = '[ckey]'")

```

Comment:
Does `ckey` need to be `sanitizeSQL`d here (and elsewhere)? `client.address`?

I guess it wasn't before, so it's probably fine.

Reply:
It's not done elsewhere so yes; the file generally lacks proper sensitization for several values.

Reply:
Neither of these can really be manipulated in any reasonable way but yeah.

Reply:
They can be VV'd